File: Array/11_20/16Solution.java
class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        Arrays.sort(nums);
        int sum=0;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
        }
        int groupSum=0;
        List<Integer> list=new ArrayList<>();
        for(int i=nums.length-1;i>=0;i--){
            groupSum+=nums[i];
            list.add(nums[i]);
            if(groupSum>sum-groupSum){
                return list;
            }
        }
        return list;
    }
}

File: Array/11_20/16Solution.py
def minSubsequence(nums):
        s = sum(nums)
        nums.sort(reverse = True)
        
        cs, i = 0, 0
        res = []
        for i, c in enumerate(nums):
            if cs > s / 2:
                return res
            cs += nums[i]
            res.append(nums[i])
            i += 1
            
        return res
nums=[ int(x.strip()) for x in input().split(",")]
print(minSubsequence(nums))